mtags/src/main/scala-3/scala/meta/internal/pc/InferredTypeProvider.scala

```diff
@@ -133,23 +134,42 @@ final class InferredTypeProvider(
           )
         }
 
+        def checkForParensAndEdit(
+            applyEndingPos: Int,
+            toCheckFor: Char,
+            blockStartPos: SourcePosition
+        ) = {
+          val text = params.text.toString()
+          val isParensFunction: Boolean = text(applyEndingPos) == toCheckFor
+
+          val alreadyHasParens =
+            text(blockStartPos.start) == '('
+
+          if (isParensFunction && !alreadyHasParens) {
+            new TextEdit(blockStartPos.toLSP, "(") :: baseEdit(withParens =
+              true
+            ) :: Nil
+          } else {
+            baseEdit(withParens = false) :: Nil
+          }
+        }
+
         def typeNameEdit: List[TextEdit] = {
           path match {
-            // lambda `{a => ??? }` block and `map{a => ??? }` apply
+            // lambda `map(a => ???)` apply
             case _ :: _ :: (block: untpd.Block) :: (appl: untpd.Apply) :: _
                 if isParam =>
-              val isParensFunction =
-                params.text.toString()(appl.fun.endPos.end) == '('
-              val alreadyHasParens =
-                params.text.toString()(block.startPos.start) == '('
-              if (isParensFunction && !alreadyHasParens)
-                new TextEdit(block.startPos.toLSP, "(") :: baseEdit(withParens =
-                  true
-                ) :: Nil
-              else
-                baseEdit(withParens = false) :: Nil
-
-            case _ => baseEdit(withParens = false) :: Nil
+              checkForParensAndEdit(appl.fun.endPos.end, '(', block.startPos)
+
+            // labda `map{a => ???}` apply
+            // Ensures that this becomes {(a: Int) => ???} since parentheses
+            // are required around the parameter of a lambda in Scala 3
+            case valDef :: defDef :: (block: untpd.Block) :: (_: untpd.Block) :: (appl: untpd.Apply) :: _
```

Comment:
I corrected the comments in here since I _think_ the one above was wrong?

For
```scala
map{a => ???}
```
It won't be caught by the case up above, so it shouldn't be in that comment right? Since this gets translated to:
```scala
map({a => ???})
```
That's where there is an extra block in the path before the apply right?

Reply:
Yeah, actually the comments were wrong. For `{}` we just annotated it with the type in `baseEdit`